cmd/gcrane/cmd/gc.go

```diff
@@ -29,28 +31,57 @@ func init() { Root.AddCommand(NewCmdGc()) }
 // NewCmdGc creates a new cobra.Command for the gc subcommand.
 func NewCmdGc() *cobra.Command {
 	recursive := false
+	before := time.Duration(0)
+	pattern := ""
+
 	cmd := &cobra.Command{
 		Use:   "gc",
 		Short: "List images that are not tagged",
 		Args:  cobra.ExactArgs(1),
 		Run: func(_ *cobra.Command, args []string) {
-			gc(args[0], recursive)
+			gc(args[0], recursive, time.Now().Add(-before), pattern)
 		},
 	}
 
 	cmd.Flags().BoolVarP(&recursive, "recursive", "r", false, "Whether to recurse through repos")
+	cmd.Flags().DurationVarP(&before, "before", "b", time.Duration(0), "Age of the oldest uploaded image to print (ns, us, ms, s, m, h)")
+	cmd.Flags().StringVarP(&pattern, "pattern", "p", "", "Will also collect images with no tags matching this pattern")
 
 	return cmd
 }
 
-func gc(root string, recursive bool) {
+func gc(root string, recursive bool, before time.Time, pattern string) {
 	repo, err := name.NewRepository(root)
 	if err != nil {
 		log.Fatalln(err)
 	}
 
 	auth := google.WithAuthFromKeychain(authn.DefaultKeychain)
 
+	re, err := regexp.Compile(pattern)
+	if err != nil {
+		log.Fatalln(err)
```

Comment:
While you're in here, do you mind pulling out these fatal logs and returning an `error` instead (just fatal in the cobra `Run` method)?

That will move us a bit closer to exposing this function.

Reply:
Done. I've taken the opportunity to use [RunE](https://github.com/spf13/cobra#returning-and-handling-errors)

Reply:
Nice. TIL.